spec/spec_helper_local.rb

```diff
@@ -1,6 +1,33 @@
-require 'rspec-puppet-facts'
-include RspecPuppetFacts
-#                                                                       # Original fact sources:
-add_custom_fact :puppetversion, Puppet.version                          # Facter, but excluded from rspec-puppet-facts
 add_custom_fact :rabbitmq_version, '3.6.1'                              # puppet-rabbitmq
 add_custom_fact :erl_ssl_path, '/usr/lib64/erlang/lib/ssl-7.3.3.1/ebin' # puppet-rabbitmq
+
+def os_specific_facts(facts)
+  case facts[:os]['family']
+  when 'Archlinux'
+    { service_provider: 'systemd', systemd: true }
+  when 'Debian'
+    case facts[:os]['release']['major']
+    when '7'
+      { service_provider: 'sysv', systemd: false }
+    when '8'
+      { service_provider: 'systemd', systemd: true }
+    when '9'
+      { service_provider: 'systemd', systemd: true }
+    when '14.04'
+      { service_provider: 'upstart', systemd: false }
+    when '16.04'
+      { service_provider: 'systemd', systemd: true }
+    when '18.04'
+      { service_provider: 'systemd', systemd: true }
+    end
+  when 'RedHat'
+    case facts[:os]['release']['major']
+    when '6'
+      { service_provider: 'sysv', systemd: false }
+    when '7'
```

Comment:
Should this just be `else`? That way you're future proof.

Reply:
fixed

Reply:
I still see this as `when '7'`. I can see arguments for both ways - but in this case, probably makes most sense to have it be `else`.